File: payloads/cpp/inode_bomb.cpp
#include <cstdio>
#include <fstream>
#include <string>

int main() {
    for (int i = 0; i < 100000; i++) {
        std::string path = "/tmp/inode_" + std::to_string(i);
        std::ofstream f(path);
        if (!f.is_open()) return 1;
        f << "x";
    }
    std::printf("ESCAPED\n");
    return 0;
}

File: payloads/java/inode_bomb.java
import java.io.File;

public class Main {
    public static void main(String[] args) throws Exception {
        for (int i = 0; i < 100000; i++) {
            new File("/tmp/inode_" + i).createNewFile();
        }
        System.out.println("ESCAPED");
    }
}
